orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/servergroup/strategies/AbstractDeployStrategyStage.groovy

```diff
@@ -97,15 +97,17 @@ abstract class AbstractDeployStrategyStage extends AbstractCloudProviderAwareSta
     Strategy strategy = (Strategy) strategies.findResult(noStrategy, {
       it.name.equalsIgnoreCase(stage.context.strategy) ? it : null
     })
+
+    def preProcessors = deployStagePreProcessors.findAll { it.supports(stage) }
     def stages = strategy.composeFlow(stage)
 
     def stageData = stage.mapTo(StageData)
-    deployStagePreProcessors.findAll { it.supports(stage) }.each {
```

Comment:
`strategy.composeFlow()` manipulates the stage context and I'd like to determine pre-processors based on the original unmodified stage

specifically the `RollingRedBlackStrategy` sets `useSourceCapacity` to `false`.